daemon/daemon.go

```diff
@@ -276,6 +277,21 @@ func (d *Daemon) Run() error {
 	close(httpReadyCh)
 	close(criReadyCh)
 
+	// init buildkit if enable builder functionality
+	//
+	// FIXME(fuweid): builder functionality is experimental version, which
+	// do not impact existing http server and cri grpc server. After it
+	// is stable, we will check the status of builder and shutdown whole
+	// daemon if the builder server is down.
+	if d.config.EnableBuilder {
+		go func() {
+			logrus.Info("serving builder server...")
+			if err := d.runBuilderServer(); err != nil {
```

Comment:
There are three type of service here, cri service, http server and builder server. A little messy to start them. We can make an abstract interface from them and manage them uniformity later.

Reply:
yeah

Reply:
if builder is enabled and start builder server failed, it is an error not a warning.

Reply:
done! thanks @yyb196